docs/src/developer_folder/contributing.rst

```diff
@@ -32,8 +32,74 @@ Finally I advise to create a dedicated conda environment for this and install Py
+
+**Code flow branches:**
+
+* **the stable branch, eg: '4.1.x'** This is the branch representing the stable version of PyMoDAQ. No change should be
+  made on this branch except bugfixes and hotfixes (see below). This is the branch from which the official releases are
+  created, for instance version 4.1.0, 4.1.1, 4.1.2, etc.
+
+* **the development branch, eg: '4.2.x_dev** Note that the branch name differs from the stable branch by one increment
+  on the minor revision number (2 instead of 1), and the '_dev' suffix is added for clarity.
+  This is the development branch. It is *ahead* of the main branch, in the sense that it contains more
+  recent commits than the main branch. It is thus the future state of the code. This is where the last developments
+  of the code of PyMoDAQ are pushed. When the developers are happy with the state of this branch, typically when they
+  finished to develop a new functionality and they tested it, this will lead to a new *release* of PyMoDAQ (4.1.x -> 4.2.0 in our example).
+  In practice, the branch will simply be renamed from *4.2.x_dev* to *4.2.x*, and a new branch *4.3.x_dev* will be created
+  to continue the cycle.
+
+**Temporary branches:**
+
+* **Feature, eg: 'feature/new_colors'**: Any additional feature should be done on a feature branch. They are created based
+  on the current development branch. When the feature is complete, a Pull Request must be open to integrate the changes into
+  the development branch.
+
+* **Bugfix, eg: 'bugfix/remove_annoying_message'**: These branches are meant to correct small issues. It can be created based
+  on either the stable or development branch, depending on where the bug is located. Regardless, any bugfix must then be applied to
+  all branches, if applicable (see note below).
+
+* **Hotfix, eg: 'hotfix/fix_huge_bug'**: This is similar to a bugfix, but for more important bugs. More precisely, hotfixes
+  are important enough that when applied, they will trigger an immediate new release (e.g. *4.1.1* -> *4.1.2*) that incorporate the fix.
+  At the contrary bugfixes can wait for a future release.
+
+.. note::
+    **Applying fixes across several branches**
+
+    Let's consider the case where a bug is found on the **stable** branch. We create a new branch to fix it, open a pull request
+    into the stable branch, and wait for it to be accepted. However, it is likely that the buggy code is also part of the
+    **development** version! To do so, we do not have to create another branch, and another pull request. If possible, we will use the
+    `rebase`__ method of git, to automatically update the dev branch from the patched stable branch. If not, we will need to
+    `cherrypick`__ the commit containing the bugfix. These are rather advanced git methods which are likely to be performed  only
+    by the maintainers, but when fixing bugs, one should always remember to check if it is present on several branches.
+
```

Comment:
concerning this, I applied a few patch these last days and because and found more convenient to apply the PR to both branches (with possibly conflict resolution for each). I would prefer this rather than rebasing because rebasing change the history of development and would cause issues for fellow developpers